homeassistant/components/modbus/switch.py

```diff
@@ -146,13 +129,24 @@ def available(self) -> bool:
         """Return True if entity is available."""
         return self._available
 
+
+class ModbusCoilSwitch(ModbusBaseSwitch):
```

Comment:
We should inherit from `SwitchEntity` somewhere in the switch entities.

Reply:
I guess the original code was written before `SwitchEntity` even existed. Anyway, the `SwitchEntity` inherits from the `ToggleEntity`. I will modify the base class to inherit directly from the `SwitchEntity`. This should solve our problem.

I will let you know.

Reply:
If the base class inherits from `SwitchEntity` we cannot inherit from the base class in other platforms than switch.

Reply:
Good point. Probably I should add the inheritance into the `ModbusCoilSwitch` and `ModbusRegisterSwitch`. This would solve our problem, and yet Fan and Light can still inherit from it.

Reply:
that would be my preference.

Reply:
Just working on a change now. I will test it real quick, then we'll run automation tests, and we should be good to go.